daemon/cluster/executor/container/executor.go

```diff
@@ -143,13 +143,22 @@ func (e *executor) Configure(ctx context.Context, node *api.Node) error {
 	attachments := make(map[string]string)
 
 	for _, na := range node.Attachments {
+		if na == nil || na.Network == nil || len(Addresses) == 0 {
+			// this should not happen, but we got a panic here and don't have a
+			// good idea about what the underlying data structure looks like.
+			logrus.WithField("NetworkAttachment", fmt.Sprintf("%#v", na)).
+				Warnf("skipping nil or malformed node network attachment entry")
+			continue
+		}
+
 		if na.Network.Spec.Ingress {
 			ingressNA = na
 		}
+
 		attachments[na.Network.ID] = na.Addresses[0]
 	}
 
-	if (ingressNA == nil) && (node.Attachment != nil) {
+	if (ingressNA == nil) && (node.Attachment != nil) && (len(ingressNA.Addresses) > 0) {
```

Comment:
Whoops; this should be `len(node.Attachment.Addresses)`; it’s failing currently:

```
21:38:52 daemon/cluster/executor/container/executor.go:146:44:warning: undeclared name: Addresses (unconvert)
21:38:52 daemon/cluster/executor/container/executor.go:146:44:warning: undeclared name: Addresses (gosimple)
21:38:52 daemon/cluster/executor/container/executor.go:146:44:warning: undeclared name: Addresses (interfacer)

Reply:
Sorry, I see my comment was a bit misleading there :blush:

Reply:
@thaJeztah the Addresses issue is at line 146. Also the len for the addresses is already being done at line 146 so not needed here.

Reply:
@fcrisciani this section is using the "old" `node.Attachment.Addresses` (singular), whereas line 146 checks for `node.Attachments[n].Addresses` (Attachments (plural))